### PR TITLE
Add basic support for tables.

### PR DESCRIPTION
**Pull request by stephenreddek**
All of the GFM Table tests pass now

### CODE REVIEW

File: test-results/failing/GFM/[extension] Tables.md [hunk 1]
```diff
@@ -84,71 +20,5 @@ Should give output:
...
-
-````````````html
-<table><thead><tr><th>abc</th><th>def</th></tr></thead></table>
-````````````
-
-But instead was:
-
-````````````html
-<p>| abc | def | | --- | --- |</p>
+<table><thead><tr><th>abc</th><th>def</th></tr></thead><tbody><tr><td>bar</td><td>baz</td></tr><tr><td>&gt; bar</td><td></td></tr></tbody></table>
```
Comment (stephenreddek) [inline test-results/failing/GFM/[extension] Tables.md:23]: I see that this is still here even though it's listed as passing in test-results/passing-GFM. Is it possible that when there are no failing tests in a file it doesn't update that file to be empty?

File: test-results/failing/New/table_cells.md [hunk 2]
```diff
@@ -72,5 +72,5 @@ Should give output:
...
...ody></table><table><thead><tr><th>1</th></tr></thead><tbody><tr><td>1\1</td></tr></tbody></table><table><thead><tr><th>1</th></tr></thead><tbody><tr><td>|</td></tr></tbody></table><table><thead><tr><th>1</th></tr></thead><tbody><tr><td>\|</td></tr></tbody></table><table><thead><tr><th>1</th><th>2</th></tr></thead><tbody><tr><td>2</td><td></td></tr></tbody></table><table><thead><tr><th>1</th><th>2</th></tr></thead><tbody><tr><td>1||2||</td><td></td></tr></tbody></table><table><thead><tr><th>1</th><th>2</th></tr></thead><tbody><tr><td></td><td>2</td></tr></tbody></table><table><thead><tr><th>1</th><th>2</th></tr></thead><tbody><tr><td>1||2|\</td><td></td></tr></tbody></table><table><thead><tr><th>1</th><th>2</th></tr></thead><tbody><tr><td></td><td>2</td></tr></tbody></table><table><thead><tr><th>1</th><th>2</th></tr></thead><tbody><tr><td>1</td><td>2|</td></tr></tbody></table><table><thead><tr><th>1</th><th>2</th></tr></thead><tbody><tr><td>1</td><td>2|</td></tr></tbody></table><table><thead><tr><th>1</th><th>2</th></tr></thead><tbody><tr><td>1</td><td>2|</td></tr></tbody></table><table><thead><tr><th>1</th><th>2</th></tr></thead><tbody><tr><td>1</td><td>2|</td></tr></tbody></table>
```
Comment (stephenreddek) [inline test-results/failing/New/table_cells.md:75]: Based on babelmark, I think this spec conflicts with GFM on how it handles `\\|`.

**Comment by stephenreddek** [comment]
@dillonkearns I was able to use the concepts we talked about to support breaking on block elements. This is ready for your review.

File: src/Markdown/Parser.elm [hunk 3]
```diff
@@ -729,15 +751,26 @@ whenPreviousWasNotBody =
...
+        |> map (\block -> Loop (possiblyMergeBlocks revStmts block))
     ]
 
 
+tableRowIfTableStarted : List RawBlock -> Parser RawBlock
+tableRowIfTableStarted revStmts =
+    case revStmts of
+        (Table (Markdown.Table.Table headers body)) :: _ ->
+            TableParser.bodyRowParser (List.length headers)
+                -- We always fill in the whole table so that we don't have an impossible state of a table row without a table header
```
Comment (stephenreddek) [inline src/Markdown/Parser.elm:767]: I know this is a bit weird when we join them by just discarding the former element, but it felt worse to have a `TableRow` `RawBlock` that wasn't guaranteed to be valid via types.

**Comment by stephenreddek** [comment]
Closes #33

**Comment by stephenreddek** [comment]
Master Benchmarks:
Screen Shot 2020-09-09 at 08 53 55
GFM Benchmarks:
Screen Shot 2020-09-09 at 08 50 36

**Comment by dillonkearns** [comment]
🤯 ❓ 

Better with the new branch? Yay! 🎉  Great stuff!

Thanks so much for seeing this through @stephenreddek, this was a big one!

**Comment by bburdette** [comment]
looking forward to trying this out!  nice work!

**Pull request merged**